src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -10,7 +10,9 @@
 
     /**
      * Parses {@code userInput} into a command and returns it.
+     *
+     * @return Command object.
      * @throws ParseException if {@code userInput} does not conform the expected format
      */
-    T parse(String userInput) throws ParseException;
+    Command parse(String userInput) throws ParseException;
```

Comment:
what's the motivation to change the generic type to Command when T extends command?